### DISCUSSION TITLE
Regressor on bias - center / demean?

### DISCUSSION THREAD

**Discussion by TinusUU** [mwshinn/PyDDM#78] [Ask for help]
Hi all, 

Suppose I have the Roitman data with an additional trial-by-trial reward-regressor 'reward' (so, on each trial, the reward is different). Now, when I want to test for a bias in drift-rate criterion (vc) I would setup the get_drift function as follows:
v = vc + driftcoh * condions['coh']

When I expect the reward regressor to modulate vc, I would setup the following:
v = vc * condions['reward'] + driftcoh * condions['coh']

However, when I want to test for a overall bias effect, with an additional modulation, I would define v: 
v = vc_global + vc*condions['reward'] + driftcoh * condions['coh']

In the latter case, I would ideally demean or 0-center the regressor, so that the intercept vc_global captures the 'overall' shift in drift-rate criterion, at the point there the modulation = 0 (or mean). 

If this is correct, wouldn't we want to de-mean the coh condition as well? 
(which was not done in [this example](https://pyddm.readthedocs.io/en/stable/cookbook/driftnoise.html?highlight=reward#biased-coherence-dependent-drift-rate) I suppose)
Or would one recommend to leave the values of both conditions / regressors what they are? 

Thanks
Martijn

**Comment by mwshinn** [comment]
PyDDM does not use R/patsy syntax.  It is normal python syntax.  So if you want reward to module the bias, you need "v = (self.vc0 if conditions["reward"] == 0 else self.vc1) + self.driftcoh*conditions["coh"]" or equivalently "v = self.vc0 + (self.vc1-self.vc0)*conditions["reward"] + self.driftcoh*conditions["coh"]", assuming reward is 0 or 1.  It does not matter whether you demean, these can be easily shown to be mathematically equivalent and will give you the same output model, albeit with different absolute magnitudes for the parameters.

**Reply by TinusUU** [reply]
Hi Max,

Thanks, but I'm not sure if I made myself clear; as described, my reward regressor is _not a dummy regressor_, but a trial-by-trial float (e.g. between -1 and 1). So, if I want to test for a **linear relation** between these trial-by-trial values **on vc**, my regressor won't fit your assumption 'reward is 0 or 1'.
  
Maybe I should have mentioned that with: 
v = vc_global + vc*condions['reward'] + driftcoh * condions['coh']

I meant something like:

```
class DriftRew(Drift):
     required_parameters = ["vc_global", "vc", "driftcoh"] 
     required_conditions = ["reward_WEIGHT", "coh"]
     def get_drift(self, conditions, **kwargs):
         return = self.vc_global + self.vc * condions['reward_WEIGHT'] + self.driftcoh * condions['coh']
```
 
Where conditions['coh'] can take values [-0.3, -0.2, -0.1, 0.1, 0.2, 0.3]
And _**_conditions['reward_WEIGHT'] can take values = random.uniform(-1,1)_**_ 

So, when we fit a model for each subject, self.vc_global will account for the individual criterion.
(note, that it is _not_ a given that criteria will be fixed for ['coh'] =0). 
While, in addition, an additional effect of reward will be added to the criterion, scaled with conditions['reward_WEIGHT']. 
In that case, if we don't demean across the random / trial-by-trial values of reward_WEIGHT, self.vc_global will have a different value / meaning, correct?

Best,
Martijn

**Reply by mwshinn** [reply]
I still don't follow what you are asking.  Demeaning will never change a linear model's predictive accuracy.  It may change the specific numbers you get out as coefficients, but it is still the same model, and you can convert from one to the other after fitting the model.  This is not related to PyDDM though, this is just a mathematical fact, so I think I still do not understand what you are asking.  How is this related to PyDDM?

**Reply by TinusUU** [reply]
Well, I guess my question isn't specifically related to pyDDM itself, but rather to setting up a correct model where the coefficients make sense given the data and prior behavioral assumptions. Since pyDDM is very flexible in setting up parameter-specific functions, I think it's a good practice to think about how the variability in the data can be explained by the parameters. And yes, demeaning will only change the coefficients and their meaning. That's exactly the point. (if we want to draw inference from individual differences in the parameters). But reading your answer, I guess it's just a manner of choice. ;-) 
Thanks.
